app/js/reducers/plugins.js

```diff
@@ -1,186 +1,34 @@
-                        input_type: 'Text',
-                        input_hint: 'Str'
-                    }
-                ],
-                output: [
-                    {
-                        type: 'Name',
-                        sub_type: 'distance_matrix',
-                        input_type: 'Text',
-                        input_hint: 'DistanceMatrix'
-                    },
-                    {
-                        type: 'Name',
-                        sub_type: 'pcoa_results',
-                        input_type: 'Text',
-                        input_hint: 'PCoAResults'
-                    }
-                ]
-            },
+import update from 'react-addons-update';
+
+const initialState = [];
+
+const pluginsReducer = (state = initialState, action) => {
+    switch (action.type) {
+    case 'FOUND_PLUGIN': {
+        const newState = [
+            ...state,
             {
-                name: 'beta-diversity',
-                info: 'Produces: DistanceMatrix',
-                description: 'Beta diversity',
-                input: [
-                    {
-                        type: 'Artifact',
-                        sub_type: 'feature-table',
-                        input_type: 'Dropdown',
-                        input_choices: 'Artifacts'
-                    },
-                    {
-                        type: 'Artifact',
-                        sub_type: 'phylogeny',
-                        input_type: 'Dropdown',
-                        input_choices: 'Artifacts'
-                    },
-                    {
-                        type: 'Parameter',
-                        sub_type: 'metric',
-                        input_type: 'Text',
-                        input_hint: 'Str'
-                    }
-                ],
-                output: [
-                    {
-                        type: 'Name',
-                        sub_type: 'DistanceMatrix',
-                        input_type: 'Text',
-                        input_hint: 'DistanceMatrix'
-                    }
-                ]
+                name: action.plugin.name,
+                workflowsURI: action.plugin.workflow_uri,
+                workflows: []
             }
-        ]
+        ];
+        return newState;
     }
-];
+    case 'FOUND_WORKFLOW': {
+        const workflow = {
+            name: action.workflow.name,
+            info: action.workflow.info,
+            description: action.workflow.description,
+            inputArtifacts: action.workflow.input_artifacts,
+            inputParameters: action.workflow.input_parameters,
+            outputArtifacts: action.workflow.output_artifacts
+        };
 
-const pluginsReducer = (state = initialState, action) => {
-    switch (action.type) {
+        const index = state.findIndex(plugin => plugin.name === action.plugin);
+        const newState = update(state, { [index]: { workflows: { $push: [workflow] } } });
```

Comment:
See if it is possible to dispatch to a child reducer to handle the workflow list, then you can just assign the new workflow list using the es7 object spread. (probably)